### DISCUSSION TITLE
How to implement micro-frontend architecture for two entirely different next js apps sharing data using local storage

### DISCUSSION THREAD

**Discussion by rishabh2712** [vercel/next.js#33014] [Help]
I have an application A that redirects to application B. But before redirecting it sets the context in local storage. User when gets back the application A, the local storage is set accordingly.

How to manage two different application in dev and prod mode for such use case

**Comment by Areg1992** [comment]
As I understood there is one big module wich is working only with webpack versio 5. It's called module @module-federation/nextjs-mf.
But the problem is that this module is not free.

**Reply by rishabh2712** [reply]
Next mf you must be talking about. The maintainer has made that open sourced recently.

**Comment by Areg1992** [comment]
Yes but the module is not free, is not opensource as I understand.
So I need to pay a lot of money to use one module.

**Reply by rishabh2712** [reply]
If incase you are talking about this.

https://github.com/module-federation/nextjs-mf/tree/main/packages/nextjs-mf

**Comment by Areg1992** [comment]
Yes, I am talking about that. Isn't there any free version to try ?
